avocado/utils/linux_modules.py

```diff
@@ -208,34 +208,91 @@ def get_loaded_modules():
         return [astring.to_text(_.split(b' ', 1)[0]) for _ in proc_modules]
 
 
-def check_kernel_config(config_name):
+def check_kernel_config(config_name, session=None):
     """
     Reports the configuration of $config_name of the current kernel
 
     :param config_name: Name of kernel config to search
+    :param session: guest session, command is run on host if None
     :type config_name: str
     :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
     :rtype: :class:`ModuleConfig`
     """
+    if session is None:
+        kernel_version = platform.uname()[2]
+        config_file = '/boot/config-' + kernel_version
+        # Sometimes a kernel config's name may contain another kernel config
```

Comment:
I am not sure I understood this comment.  Is there supposed to be any built-in logic here that will automatically refer to another config entry?

Reply:
> I am not sure I understood this comment. Is there supposed to be any built-in logic here that will automatically refer to another config entry?

This comment aims to  explain why I  use `grep ${config_name}= $config_file`  to get config info.
The '${config_name}=' could avoid getting multiple config.
E.g. When we check CONFIG_PREEMPT, there are multiple kernel configs contain 'CONFIG_PREEMPT'
```
# grep CONFIG_PREEMPT $config_file
CONFIG_PREEMPT_BUILD=y
# CONFIG_PREEMPT_NONE is not set
CONFIG_PREEMPT_VOLUNTARY=y
# CONFIG_PREEMPT is not set
CONFIG_PREEMPT_COUNT=y
CONFIG_PREEMPTION=y
CONFIG_PREEMPT_DYNAMIC=y
CONFIG_PREEMPT_RCU=y
CONFIG_PREEMPT_NOTIFIERS=y
# CONFIG_PREEMPT_TRACER is not set
# CONFIG_PREEMPTIRQ_DELAY_TEST is not set
```

Reply:
OK, now I get it.  I had understood it like if there was a reference another configuration item... but you meant common prefixes and the need to be deterministic by using the `=` sign.

BTW, I believe matching text is common enough that the code itself (without the comment) can be more obvious than with the comment (the existence of the comment may lead one into believing that it's something out of the ordinary).

Reply:
> OK, now I get it. I had understood it like if there was a reference another configuration item... but you meant common prefixes and the need to be deterministic by using the `=` sign.
> 
> BTW, I believe matching text is common enough that the code itself (without the comment) can be more obvious than with the comment (the existence of the comment may lead one into believing that it's something out of the ordinary).

Sorry for the ambiguity. The comment was removed in the latest PR.